### PR TITLE
Add some type hints to tests files

### PR DESCRIPTION
**Pull request by dklimpel**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

Signed-off-by: Dirk Klimpel dirk@klimpel.org

### CODE REVIEW

File: tests/storage/test_base.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def runWithConnection(func, *args, **kwargs):
         db = DatabasePool(Mock(), Mock(config=sqlite_config), fake_engine)
         db._db_pool = self.db_pool
 
-        self.datastore = SQLBaseStore(db, None, hs)
+        self.datastore = SQLBaseStore(db, None, hs)  # type: ignore[arg-type]
```
Comment (dklimpel) [inline tests/storage/test_base.py:63]: error:
`tests/storage/test_base.py:63: error: Argument 2 to "SQLBaseStore" has incompatible type "None"; expected "LoggingDatabaseConnection"  [arg-type]`

File: tests/storage/test_roommember.py [hunk 2]
```diff
@@ -34,7 +34,7 @@ class RoomMemberStoreTestCase(unittest.HomeserverTestCase):
         room.register_servlets,
     ]
 
-    def prepare(self, reactor: MemoryReactor, clock: Clock, hs: TestHomeServer) -> None:
+    def prepare(self, reactor: MemoryReactor, clock: Clock, hs: TestHomeServer) -> None:  # type: ignore[override]
```
Comment (dklimpel) [inline tests/storage/test_roommember.py:37]: error:
`tests/storage/test_roommember.py:37: error: Argument 3 of "prepare" is incompatible with supertype "HomeserverTestCase"; supertype defines the argument type as "HomeServer"  [override]`

File: changelog.d/12833.misc [hunk 3]
```diff
@@ -0,0 +1 @@
+Add some type hints to tests files.
```
Comment (anoadragon453) [inline changelog.d/12833.misc:1]:
```suggestion
Add some type hints to test files.
```

**Review by anoadragon453 — APPROVED** [review]
Looks good to me. Thank you very much!

**Pull request merged**